Application/Client/src/Admin.cpp

```diff
@@ -0,0 +1,294 @@
+#include <iostream>
+#include <limits>
+#include <stdexcept>
+
+#include "Admin.h"
+#include "Utils.h"
+#include "RequestType.h"
+
+enum class AdminOptions {
+    DISPLAY_MENU,
+    VIEW_SPECIFIC_DATE_MENU,
+    VIEW_FEEDBACK_RATINGS,
+    ADD_MENU_ITEM,
+    REMOVE_MENU_ITEM,
+    ADD_EMPLOYEE,
+    REMOVE_EMPLOYEE,
+    VIEW_DISCARDED_MENU_ITEM,
+    LOGOUT
+};
+
+void Admin::adminLandingPage() {
+    try {
+        std::vector<std::string> options = {"Display Menu Items", "View Specific Date Menu", "View Feedback & Ratings", "Add Menu Item", "Remove Menu Item", "Add Employee", "Remove Employee", "View Discarded Menu Item", "Logout"};
+
+        std::cout << "Press enter to continue: ";
+        std::cin.ignore();
+        int selected_option = Utils::selectOption(options);
+        
+        switch(static_cast<AdminOptions>(selected_option)) {
+            case AdminOptions::DISPLAY_MENU:
+                displayMenu();
+                break;
+            case AdminOptions::VIEW_SPECIFIC_DATE_MENU:
+                viewSpecificDateMenu();
+                break;
+            case AdminOptions::VIEW_FEEDBACK_RATINGS:
+                viewFeedbackRatings();
+                break;
+            case AdminOptions::ADD_EMPLOYEE:
+                handleAddEmployee();
+                break;
+            case AdminOptions::REMOVE_EMPLOYEE:
+                handleRemoveEmployee();
+                break;
+            case AdminOptions::ADD_MENU_ITEM:
+                handleAddMenuItem();
+                break;
+            case AdminOptions::REMOVE_MENU_ITEM:
+                handleRemoveMenuItem();
+                break;
+            case AdminOptions::VIEW_DISCARDED_MENU_ITEM:
+                viewDiscardedMenuItems();
+                break;
+            case AdminOptions::LOGOUT:
+                std::cout << "Logging out..." << std::endl;
+                exit(0);
+            default:
+                std::cout << "Invalid Option" << std::endl;
+                break;
+        };
+    } catch (const std::exception& e) {
+        std::cerr << "An error occurred: " << e.what() << std::endl;
+    } catch (...) {
+        std::cerr << "An unknown error occurred" << std::endl;
+    }
+}
+
+void Admin::handleAddEmployee() {
+    try {
+        std::string employeeDetails = getEmployeeDetails();

```

Comment:
This could be moved out of the function and passed to this function to maintain SRP as discussed in our session.